### PR TITLE
Sample for imagenet and batch prediction

### LINKED ISSUE #853: Need sample or document about batch predict
There's C API Sample so that I can usnig this lib in C API,but the C API Sample only provide predict once a time,I need one more Sample or document about batch predict.

### PR DESCRIPTION
**Pull request by snnn**
**Description**: Describe your changes.
A sample application for reading imagenet validation data(50000 jpeg images) as input, run prediction on them and calculate the TOP-1 accuracy.

**Motivation and Context**
Create a sample application for the C API.

- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.
To resolve #853

### CODE REVIEW

File: samples/c_cxx/imagenet/run_release.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+LD_LIBRARY_PATH=/home/chasun/src/onnxruntime/cmake/cmake-build-relwithdebinfo ../cmake-build-relwithdebinfo/imagenet/image_classifier /home/chasun/src/imagnet_validation_data "/home/chasun/src/onnxruntime/scripts/opset_10/inception_v4/model.onnx" /home/chasun/src/onnxruntime/samples/c_cxx/imagenet/imagenet_lsvrc_2015_synsets.txt /home/chasun/src/onnxruntime/samples/c_cxx/imagenet/imagenet_2012_validation_synset_labels.txt
```
Comment (jignparm) [inline samples/c_cxx/imagenet/run_release.sh]: Make "/home/chasun/src/onnxruntime" more generic/relative.

File: samples/c_cxx/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ string(APPEND CMAKE_CXX_FLAGS " /W4")
+else()
+ string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra")
+ string(APPEND CMAKE_C_FLAGS " -Wall -Wextra")
+endif()
+
+message("${CMAKE_SYSTEM_INCLUDE_PATH}")
+find_package(JPEG REQUIRED)
+find_package(PNG)
+
```
Comment (jignparm) [inline samples/c_cxx/CMakeLists.txt]: Is the JPEG package automatically installed? Should we include the build command as part of a README.md? It should be automatic to get started with the  examples once ORT is build (e.g. similar to MNIST).

```
CMake Error at D:/cmake3.13.1/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR)
```

File: samples/c_cxx/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+add_definitions(-DUSE_CUDA)
+
+#TODO: remove this part(need to talk to Ryan and find a solution for this)
+if(UNIX)
+ include_directories("/usr/local/include/onnxruntime")
+else()
+ include_directories("C:\\Program Files (x86)\\onnxruntime\\include" "C:\\Program Files (x86)\\onnxruntime\\include\\onnxruntime" "C:\\Program Files (x86)\\onnxruntime\\include\\onnxruntime\\core\\session")
+ link_directories("C:\\Program Files (x86)\\onnxruntime\\lib")
```
Comment (jignparm) [inline samples/c_cxx/CMakeLists.txt]: There's a `TODO` above -- assuming the hard-coded directories are supposed to be removed before merge.

File: samples/c_cxx/CMakeLists.txt (hunk 2 above)
Comment (snnn) [reply samples/c_cxx/CMakeLists.txt]: > Is the JPEG package automatically installed

No. People need to manually install it.
Now I changed it as optional, so it won't be an issue anymore.

File: samples/c_cxx/CMakeLists.txt (hunk 3 above)
Comment (snnn) [reply samples/c_cxx/CMakeLists.txt]: Hi @jignparm, sorry, I'll solve it in another PR.

**Pull request merged**